broadinstitute/firecloud-orchestration
# [NOT YET] Large TSV Downloads

NOT READY YET!
Addresses https://broadinstitute.atlassian.net/browse/GAWB-1998

### Changes in this PR:
* Added a streaming actor that does not use the PerRequest pattern
* Use paginated endpoint from Rawls to get all entities
* Added a async capabilities to the HttpRawlsDAO

### Testing:
* Generate a workspace with lots of entities. I have a groovy script that does this - touch base with me and we can make some participant files of any size we need. 
* Export entities from data tab.

---

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
